### PR TITLE
blas: A "dummy package" to allow users to use the system -lblas and -…

### PR DESCRIPTION
**Pull request by BarrySmith**
…lapack

A "dummy" package that allows one to use the system blas and lapack without contaminating the
rpath with system directories. It sets up the spack directories with links back to the system directories

Funded-by: IDEAS
Project: IDEAS/xSDK
Time: 1 hour

### CODE REVIEW

**Comment by tgamblin** [comment]
@BarrySmith: see #2083 -- we actually just added in filtering of system paths and removed the part about "false paths" from the docs, so you don't have to add weird paths in `packages.yaml` anymore.

Given that, does it make sense to add a generic BLAS/LAPACK like this?  Implementations like `MKL` have differently named libs, and the user won't have detailed provenance in `spec.yaml` once the build is done.

**Comment by davydden** [comment]
and it is also documented already, see https://github.com/LLNL/spack/pull/2081

File: var/spack/repos/builtin/packages/system-blas-lapack/package.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+class SystemBlasLapack(Package):
+    """Allows one to use the system installed blas and lapack
+    without contaminating the rpath with system directories
+
+    """
+    homepage = "http://www.netlib.org/lapack/"
+    # Dummy url since Spack complains if I don't list something, will be
+    # removed when I don't need to download a dummy file
+    url      = 'https://bitbucket.org/saws/saws/get/master.tar.gz'
+
```
Comment (davydden) [inline var/spack/repos/builtin/packages/system-blas-lapack/package.py]: there should be a variant `shared`, `find_libraries` should then use it. That is done in all other Blas/Lapack to control which libraries to give to dependents.

File: var/spack/repos/builtin/packages/system-blas-lapack/package.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+class SystemBlasLapack(Package):
+    """Allows one to use the system installed blas and lapack
+    without contaminating the rpath with system directories
+
+    """
+    homepage = "http://www.netlib.org/lapack/"
+    # Dummy url since Spack complains if I don't list something, will be
+    # removed when I don't need to download a dummy file
+    url      = 'https://bitbucket.org/saws/saws/get/master.tar.gz'
```
Comment (davydden) [inline var/spack/repos/builtin/packages/system-blas-lapack/package.py]: probably Spack does not yet have functionality to install packages which actually don't download anything and have no sources, they only symlink things like what happens below.

**Comment by davydden** [comment]
@BarrySmith :  Could you please try using the documented/recommended approach and see if you encounter any issues (I hope you won't)? If all works fine for you, we could then close this PR.

**Comment by citibeth** [comment]
IF this PR is merged... I think we need to officially support bundle packages first.  We can't have dummy tarballs proliferating randomly....

**Comment by BarrySmith** [comment]
The updated documented/recommended approach seems to work fine so I am closing this pull request as unneeeded. If I have problems with the updated documented/recommended approach I will report issues off of that. Thanks.

**Pull request closed without merging**